### PR TITLE
refactor: refactor enroll endpoint

### PR DESCRIPTION
**Pull request by pyphilia**
Create EnrollService to be able to use repositories when needed (get item without permission, create membership without permission).

Next steps would be to remove `throwOnForbiddenPermission` for get items in ItemService.

Added some `private`.

### CODE REVIEW

**Review by MatveyK — APPROVED** [review]
LGTM ! I really like what I'me seeing here. I much prefer this division of responsibilities between service and the controller. Also the removal of the `throwOnForbiddenPermission` is very pleasant to me.

File: src/services/item/plugins/enroll/service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  CannotEnrollItemWithoutItemLoginSchema,
+} from '../../../itemLogin/errors';
+import { ItemLoginService } from '../../../itemLogin/service';
+import { ItemMembershipAlreadyExists } from '../../../itemMembership/plugins/MembershipRequest/error';
+import { Member } from '../../../member/entities/member';
+import { ItemService } from '../../service';
+
+@singleton()
+export class EnrollService {
+  private itemService: ItemService;
```
Comment (ReidyT) [inline src/services/item/plugins/enroll/service.ts:18]:
```suggestion
```

File: src/services/item/plugins/enroll/service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  private itemLoginService: ItemLoginService;
+
+  constructor(itemService: ItemService, itemLoginService: ItemLoginService) {
+    this.itemService = itemService;
+    this.itemLoginService = itemLoginService;
+  }
```
Comment (ReidyT) [inline src/services/item/plugins/enroll/service.ts:24]:
```suggestion
  constructor(itemLoginService: ItemLoginService) {
    this.itemLoginService = itemLoginService;
  }
```

**Review by ReidyT — APPROVED** [review]
Really nice changes, LGTM 🦄

**Review by spaenleh — APPROVED** [review]
Looks good like this. Just apply suggestion by @ReidyT to remove the dependency on the itemService if you don't use it.

**Pull request merged**